### PR TITLE
add test for odo url create --context flag

### PR DESCRIPTION
**Pull request by girishramnani**
**What type of PR is this?**
> Uncomment only one ` /kind` line, and delete the rest.
> For example, `> /kind bug` would simply become: `/kind bug`

/kind bug

**What does does this PR do / why we need it**:
See $SUBJECT

**Which issue(s) this PR fixes**:

No issue

**PR acceptance criteria**:

- [ ] Unit test 

- [X] Integration test 

- [ ] Documentation 

- [X] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:
test should pass

### CODE REVIEW

**Comment by dharmit** [comment]
Works for me.

/approve

File: pkg/odo/cli/url/create.go [hunk 1]
```diff
@@ -97,8 +97,8 @@ func NewURLCreateOptions() *URLCreateOptions {
 }
 
 // Complete completes URLCreateOptions after they've been Created
-func (o *URLCreateOptions) Complete(_ string, cmd *cobra.Command, args []string) (err error) {
-	o.DevfilePath = clicomponent.DevfilePath
+func (o *URLCreateOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
+	o.PushOptions.CompleteDevfilePath()
```
Comment (mik-dass) [inline pkg/odo/cli/url/create.go:101]: We can just do `o.CompleteDevfilePath()` like done here https://github.com/openshift/odo/pull/3857

**Comment by mik-dass** [comment]
PR needs rebase
/hold

File: pkg/odo/cli/url/create.go (hunk 1 above)
Comment (girishramnani) [reply pkg/odo/cli/url/create.go]: addressed

**Comment by girishramnani** [comment]
```
msg="Error: [ERR]: Error building changeset: InvalidChangeBatch: [Tried to create resource record set [name='api.ci-op-mkvnyjxp-2f74e.origin-ci-int-aws.dev.rhcloud.com.', type='A'] but it already exists]"
level=error msg="\tstatus code: 400, request id: 3fcb7c32-1215-42b6-ac46-ff1df0b4999f"
level=error
level=error msg="  on ../tmp/openshift-install-253236301/route53/base.tf line 33, in resource \"aws_route53_record\" \"api_external\":"
level=error msg="  33: resource \"aws_route53_record\" \"api_external\" {"
level=error
level=error
level=fatal msg="failed to fetch Cluster: failed to generate asset \"Cluster\": failed to create cluster: failed to apply Terraform: failed to complete the change"
error: failed to execute wrapped command: exit status 1
```
infra issue
/retest

File: tests/integration/devfile/cmd_devfile_url_test.go [hunk 2]
```diff
@@ -182,22 +196,14 @@ var _ = Describe("odo devfile url command tests", func() {
...
 			stdOut := helper.CmdShouldFail("odo", "url", "create", "--tls-secret", "foo", "--port", "3000", "--ingress")
 			Expect(stdOut).To(ContainSubstring("TLS secret is only available for secure URLs of Ingress kind"))
-		})
-
-		It("should report multiple issues when it's the case", func() {
-			helper.CmdShouldPass("odo", "create", "nodejs", "--project", namespace)
-			stdOut := helper.CmdShouldFail("odo", "url", "create", "--host", "https://127.0.0.1:60104", "--tls-secret", "foo", "--port", "3000", "--ingress")
+			stdOut = helper.CmdShouldFail("odo", "url", "create", "--host", "https://127.0.0.1:60104", "--tls-secret", "foo", "--ingress")
 			Expect(stdOut).To(And(ContainSubstring("is not a valid host name"), ContainSubstring("TLS secret is only available for secure URLs of Ingress kind")))
+			stdOut = helper.CmdShouldFail("odo", "url", "create", "--host", "https://127.0.0.1:60104", "--ingress")
```
Comment (mik-dass) [inline tests/integration/devfile/cmd_devfile_url_test.go:205]: tests are failing as we need to provide the port here https://travis-ci.com/github/openshift/odo/jobs/383119575

File: tests/integration/devfile/cmd_devfile_url_test.go [hunk 3]
```diff
@@ -48,20 +48,34 @@ var _ = Describe("odo devfile url command tests", func() {
...
+			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile.yaml"), filepath.Join(subFolderContext, "devfile.yaml"))
+
+			stdout := helper.CmdShouldFail("odo", "url", "list", "--context", subFolderContext)
+			helper.MatchAllInOutput(stdout, []string{
+				"no URLs found",
+				"Refer `odo url create -h` to add one",
+			})
+
+			stdout = helper.CmdShouldFail("odo", "url", "create", url1, "--port", "8080", "--context", subFolderContext)
+			Expect(stdout).To(ContainSubstring("is not exposed"))
```
Comment (mik-dass) [inline tests/integration/devfile/cmd_devfile_url_test.go:70]: tests are failing here https://travis-ci.com/github/openshift/odo/jobs/383119575#L1779

**Review by mik-dass — APPROVED** [review]
/lgtm

**Pull request merged**